library/src/main/java/com/google/android/exoplayer2/extractor/mkv/MatroskaExtractor.java

```diff
@@ -1608,4 +1628,118 @@ private static boolean parseMsAcmCodecPrivate(ParsableByteArray buffer) throws P
 
   }
 
+  private class TrueHDSampleHolder {
```

Comment:
latest changes get rid of the extra copying.  I trimmed the every 16 samples down to 8 to make it more responsive, but when you get lower the shield tv keeps getting buffer underruns.  The shield tv also has problems if the first data sent to its truehd passthrough decoder isn't a sync frame.  I'll look into it further if I can do a better job and get the encoded frame size.